rust/tools/authorization-logic/src/souffle/lowering_ast_datalog.rs

```diff
@@ -314,6 +322,26 @@ impl LoweringToDatalogPass {
             .flatten()
             .collect()
     }
+    
+    fn rvalue_to_dlir(
+        &mut self,
+        speaker: &AstPrincipal,
+        rvalue: &AstRValue) -> DLIRRvalue {
+        match rvalue {
+            AstRValue::FlatFactRValue { flat_fact } => {
+                let (flat_pred, _) = self.flat_fact_to_dlir(&flat_fact, &speaker);
+                pred_to_dlir_rvalue(&push_prin(String::from("says_"),
+                    &speaker, &flat_pred))
+            }
+            AstRValue::BinopRValue { lnum, binop, rnum } => {
+                DLIRRvalue::BinopRValue { 
+                    lnum: lnum.clone(), 
+                    binop: *binop, 
+                    rnum: rnum.clone() 
+                }
+            }
```

Comment:
Note that if `AstRValue` and `DLIRRvalue` shared the same subclass, this would just be something like:

```Rust
AstRValue::BinOp(binop_rval) => DLIRRvalue::BinopRValue { binop: binop_rval.clone() }
```

Reply:
There's no subclasses in Rust, just traits.

Reply:
Sorry, not subclass, `struct` embedded within analogous branch of the `enum`.

Reply:
OK I get it now thanks. I took your other suggestion which is relevant and this is fixed now.